File: code/283.swift
class Solution {
    func moveZeroes(inout nums: [Int]) {
        var j = 0
        var count = nums.count
        while(j<count) {
            if nums[j] == 0 {
                nums.removeAtIndex(j)
                nums.append(0)
                count -= 1
            }else{
                j += 1;
            }
        }
    }
}

File: code/283.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        for(int lastFoundAtNoneZero=0,cur=0; cur<nums.size(); cur++) {
            if(nums[cur] != 0){
                while(nums[lastFoundAtNoneZero]!=0 && lastFoundAtNoneZero<cur){
                    lastFoundAtNoneZero++;
                }
                swap(nums[lastFoundAtNoneZero++],nums[cur]);
            }
        }
        
    }
};
